core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -1974,27 +1982,42 @@ private static class TimestampDiffConvertlet implements SqlRexConvertlet {
       SqlIntervalQualifier qualifier;
       final RexNode op1;
       final RexNode op2;
-      if (call.operand(0).getKind() == SqlKind.INTERVAL_QUALIFIER) {
-        qualifier = call.operand(0);
-        op1 = cx.convertExpression(call.operand(1));
-        op2 = cx.convertExpression(call.operand(2));
-      } else {
+      final boolean isBigQuery = !(call.operand(0).getKind() == SqlKind.INTERVAL_QUALIFIER);
```

Comment:
I don't think the `isBigQuery` variable helps much. You should extend the comment, a few lines previously, about the two variants, and how one only occurs in BigQuery mode.

You should also explain why TIMESTAMP_TRUNC calls are necessary.

Reply:
I moved and modified my comments a bit, feel free to let me know if you have other suggestions.